### ISSUE TITLE
How to make bars filled with more than two colors correctly in anno_numeric?

### ISSUE THREAD

**Issue by guodudou** [jokergoo/ComplexHeatmap#1031]
ginfo:
![image](https://user-images.githubusercontent.com/47578531/208745903-e4882b9c-aadf-4a9f-9b85-1c9b1785e0e6.png)

```
anno209 <- rowAnnotation(
    log2fc1 = anno_numeric(ginfo$log2FC209 %>% nice_round(., digit=3), 
                           bg_gp = gpar(fill = ginfo$col209),
                           labels_gp = gpar(fontsize = letter_size),
                           bar_width = unit(1, "npc") - unit(2, "pt")), 
    annotation_name_rot = row_anno_rot,
    annotation_name_gp = gpar(fontsize = label_size, fontface = "bold"),
    width=unit(c(row_anno_width), "in"),
    annotation_name_offset = unit(c(row_anno_offset), "mm"),
    annotation_label = "log2FC 209_5 VS 313_1" 
  );
```

I intended to have log2FC barplot as a row annotation using anno_numeric, fill the bars that are significantly (padj209 <= 0.05) upregulated (log2FC209 > 0) red, significantly (padj209 <= 0.05) downregulated  (log2FC209 < 0) green, not significant (padj209 > 0.05) white. Based on the criteria, I have the color ready as col209. Using the code above, I got bar annotation as below:

![image](https://user-images.githubusercontent.com/47578531/208747267-5f6871b8-6d8a-4c8a-b44a-acc69bf0323a.png)

Basically, it failed to add the color appropriately. Do you know how to fix it?

Thank you very much,
Wendy

**Comment by RobertWhitener** [comment]
Hi Wendy,

I am not the developer of this package, just a user.

But looking at this specific annotation, it looks like the anno_numeric(bg_gp(fill =)) option probably only accepts two color values... the first would be for negative values and the second would be for positive values.

Looking at ginfo$col209, both the first and second entries are "white", which may explain why everything is colored white.

Have you tried 
bg_gp = gpar(fill = c('green','red') 

Just to see if you get any color at all?

Robert

**Comment by guodudou** [comment]
>

**Comment by guodudou** [comment]
@RobertWhitener Hi Robert, thanks for your suggestion. Yes, I can get colors, but the color doesn't match what I want. You are right, seems like we can only use two colors in "anno_numeric" correctly. I hope the developer can help figure out how to add more than two colors which will be really helpful in this type of annotation.  

Best,
Wendy

**Comment by guodudou** [comment]
@jokergoo @RobertWhitener Hi Robert, thanks for your suggestion. Yes, I can get colors, but the color doesn't match what I want. You are right, seems like we can only use two colors in "anno_numeric" correctly. I hope the developer can help figure out how to add more than two colors which will be really helpful in this type of annotation.

Best,
Wendy
